src/g3log/logmessage.hpp

```diff
@@ -73,7 +76,7 @@ namespace g3 {
       LogMessage& operator=(LogMessage other);
 
 
-      LogMessage(const std::string& file, const int line, const std::string& function, const LEVELS& level);
+      LogMessage(const char * file, const int line, const std::string& function, const LEVELS& level);
```

Comment:
Unfortunately you cannot rely on the `char*` to remain valid across library borders. That's why the data is copied into the class. One runtime library can delete the data in a different way than expected in the LogMessage receiving runtime.  

It is *probably* safe to change this but I know it works when copied so please keep the copy paradigm. i.e. keeping it as std::string.